examples/langchain-langserve/app/message_history_vector_store_chain.py

```diff
@@ -103,15 +107,22 @@ def _combine_documents(
     return document_separator.join(doc_strings)
 
 
-_search_query = RunnableLambda(
-    lambda x: zep.memory.synthesize_question(session_id=x["session_id"]),
-)
+async def _search_query(session_id: str) -> str:
+    question = zep.memory.synthesize_question(session_id=session_id)
+    if question == "":
+        return ""
+
+    documents = await retriever.ainvoke(input=question, session_id=session_id)
+
+    return _combine_documents(documents)
+
+
 _inputs = RunnableParallel(
     {
         "question": lambda x: x["question"],
         "session_id": lambda x: x["session_id"],
```

Comment:
@danielchalef Lgtm in terms of readability and transparency of what is happening, but how does this improve performance? (Beyond handling empty question case)?